tests/testImageMapReduce.py

```diff
@@ -0,0 +1,243 @@
+from __future__ import absolute_import, division, print_function
+from builtins import range
+from past.builtins import basestring
+#
+# LSST Data Management System
+# Copyright 2016 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+
+import unittest
+import numpy as np
+
+import lsst.utils.tests
+import lsst.afw.image as afwImage
+import lsst.afw.geom as afwGeom
+import lsst.pex.config as pexConfig
+import lsst.meas.algorithms as measAlg
+from lsst.ip.diffim.imageMapReduce import (ImageMapReduceTask, ImageMapReduceConfig,
+                                         ImageMapperSubtask, ImageMapperSubtaskConfig,
+                                         ImageReducerSubtask, ImageReducerSubtaskConfig)
+
+
+def setup_module(module):
+    lsst.utils.tests.init()
+
+
+class TestImageMapperSubtaskConfig(ImageMapperSubtaskConfig):
+    """!
+    \anchor TestImageMapperSubtaskConfig_
+
+    \brief Configuration parameters for the TestImageMapperSubtask
+    """
+    addAmount = pexConfig.Field(
+        dtype=float,
+        doc="""Amount to add to image""",
+        default=10.
+    )
+
+
+class TestImageMapperSubtask(ImageMapperSubtask):
```

Comment:
Prefacing a non-test class with `Test` is confusing. Maybe call it `AddAmountImageMapperSubtask`, since that's what it actually does, and note in the class docstring that it's a simple `ImageMapperSubtask` to test ImageMapper functionality? Similarly for its Config, and the other MapReduceSubtasks.

Now that I've looked at the testing code, renaming them to something closer to their functionality makes even more sense: I can't tell TestStuff1 from TestStuff2, but for one you test that an addition happened, and for the other you test a mean.